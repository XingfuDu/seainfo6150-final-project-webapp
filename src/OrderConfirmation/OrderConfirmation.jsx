import React from 'react';
import classes from './OrderConfirmation.module.css';

const OrderConfirmation = (props) => {
	const query = new URLSearchParams(props.location.search);
	const form = [];
	for (let param of query.entries()) {
		form.push(param[0]+':  '+param[1]);
	}
	return (
		<div className={classes.wrapper}>
			<article className={classes.article}>
				<h3 className={classes.h3}>Hello,</h3>
				<p className={classes.p}>Thank you for shopping with us! Your order details are indicated below.</p>
                <div className={classes.form}>
                    {form.map(data=><p className={classes.p}>{data}</p>)}
                </div>
			</article>
		</div>
	);
};

export default OrderConfirmation;
