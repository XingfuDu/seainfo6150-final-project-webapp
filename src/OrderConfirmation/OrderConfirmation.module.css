.wrapper {
	width: 100%;
	min-height: 100%;
	background-color: #f2e9e4;
	padding: 15px;
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
}
.article {
	min-height: 100%;
	font-family: 'Lora', serif;
	font-size: 150%;
}

.h3 {
	color: #ee6c4d;
}

.p {
	color: #3d5a80;
}

.form {
    display: flex;
    flex-direction: column;
    margin-top: 50px;
    border: 1px solid #eee;
    box-shadow: 0 2px 3px #ccc;
    padding: 40px;
}
