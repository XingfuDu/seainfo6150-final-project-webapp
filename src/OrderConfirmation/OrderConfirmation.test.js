import React from 'react';
import OrderConfirmation from './OrderConfirmation';
import { render } from "@testing-library/react";

describe('OrderConfirmation tests', () => {
	const location = {
		search : "Book=Remember%3A%20The%20Science%20of%20Memory%20and%20the%20Art%20of%20Forgetting&Quantity=1&TotalPrice=13.52%24&Name=Xingfu%20Du&Street=3100%20Lake%20Brook%20Boulevard&State=TN&ZipCode=37909&Email=xdu5%40vols.utk.edu&DeliveryMethod=fastest&ReceiveEmails="
	}
	it('renders correctly', () => {
		const { container } = render(<OrderConfirmation location={location} />);
		expect(container).toMatchSnapshot();
	});
});
