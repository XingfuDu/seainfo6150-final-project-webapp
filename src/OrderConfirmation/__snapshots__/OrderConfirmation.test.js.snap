// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderConfirmation tests renders correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <article
      class="article"
    >
      <h3
        class="h3"
      >
        Hello,
      </h3>
      <p
        class="p"
      >
        Thank you for shopping with us! Your order details are indicated below.
      </p>
      <div
        class="form"
      >
        <p
          class="p"
        >
          Book:  Remember: The Science of Memory and the Art of Forgetting
        </p>
        <p
          class="p"
        >
          Quantity:  1
        </p>
        <p
          class="p"
        >
          TotalPrice:  13.52$
        </p>
        <p
          class="p"
        >
          Name:  Xingfu Du
        </p>
        <p
          class="p"
        >
          Street:  3100 Lake Brook Boulevard
        </p>
        <p
          class="p"
        >
          State:  TN
        </p>
        <p
          class="p"
        >
          ZipCode:  37909
        </p>
        <p
          class="p"
        >
          Email:  xdu5@vols.utk.edu
        </p>
        <p
          class="p"
        >
          DeliveryMethod:  fastest
        </p>
        <p
          class="p"
        >
          ReceiveEmails:  
        </p>
      </div>
    </article>
  </div>
</div>
`;
